### PR TITLE
Gelu Backward, Contribution from Kevin Stephano

### PR DESCRIPTION
**Pull request by Krovatkin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58249 Gelu Backward, Contribution from Kevin Stephano**
* #57750 hardtanh
* #57749 init hardshrink
* #57828 enable grad check

Differential Revision: [D28425629](https://our.internmc.facebook.com/intern/diff/D28425629)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1543,6 +1543,12 @@ def sample_inputs_hardswish(self, device, dtype, requires_grad):
                requires_grad=requires_grad, low=-5, high=5)) for _ in range(1, N)]
     return tensors
 
+def sample_inputs_gelu(self, device, dtype, requires_grad):
```
Comment (eellison) [inline torch/testing/_internal/common_methods_invocations.py:1546]: again, dont think this is actually needed but can be removed in a follow up

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -5011,6 +5017,15 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
            autodiff_nonfusible_nodes=["aten::hardtanh"]),
+    OpInfo('nn.functional.gelu',
+           aten_name="gelu",
+           supports_autograd=True,
+           assert_autodiffed=True,
+           sample_inputs_func=sample_inputs_gelu,
+           dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
+           supports_gradgrad=True,
+           supports_out=False,
+           autodiff_nonfusible_nodes=["aten::gelu"]),
```
Comment (eellison) [inline torch/testing/_internal/common_methods_invocations.py:5028]: same with this, lets remove this in follow up

**Review by eellison — APPROVED** [review]

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425629).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425629).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425629).

**Pull request closed without merging**